onmt/Factory.lua

```diff
@@ -244,6 +253,16 @@ function Factory.buildGenerator(rnnSize, dicts)
   end
 end
 
+function Factory.buildAttention(args)
```

Comment:
OK, we are indeed doing it for encoders.

However, I still think the attention module should actually be built here (otherwise the function name is ambiguous as just the class is returned).